### PR TITLE
tests: retry mounting the udisk2 device due to timing issue

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The test fails due to a timing issue (with a sleep before the mount the
error is not reproduced anymore).

error: https://travis-ci.org/snapcore/snapd/jobs/645851946#L12371

### CODE REVIEW

File: tests/main/interfaces-udisks2/task.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ details: |
     The udisks2 interface allows operating as or interacting with the UDisks2 service
 
 # Interfaces not defined for ubuntu core systems
-# FIXME: fails in debian-sid for unknown reasons
-systems: [-ubuntu-core-*, -debian-sid-*]
+# FIXME: fails in debian-sid and ubuntu-20.04 for unknown reasons
+systems: [-ubuntu-core-*, -debian-sid-*, -ubuntu-20.04-*]
```
Comment (stolowski) [inline tests/main/interfaces-udisks2/task.yaml]: Can we have a short retry loop around 'test-snapd-udisks2.udisksctl mount ..' instead? Apparently there is a race with the device becoming registered by udisks2? Alternatively, a retry loop around "test-snapd-udisks2.udisksctl  dump | MATCH $device" before mounting it, to wait for the device to be visible for udisks.

Comment (sergiocazzolato) [reply tests/main/interfaces-udisks2/task.yaml]: Sure, thanks for the suggestion

Comment (sergiocazzolato) [reply tests/main/interfaces-udisks2/task.yaml]: @stolowski the test has been fixed, thanks for the suggestion

File: tests/main/interfaces-udisks2/task.yaml [hunk 2]
```diff
@@ -41,7 +41,9 @@ execute: |
 
     device="$(losetup -j "$FS_PATH" | cut -d: -f1)"
 
-    test-snapd-udisks2.udisksctl mount -b "$device" -t ext4 | MATCH "Mounted /dev/"
+    # We retry because there is a race with the device becoming
+    # registered by udisks2. The issue can be easy reproduced in ubuntu-20.04
+    retry-tool -n 3 --wait 1 test-snapd-udisks2.udisksctl mount -b "$device" -t ext4 | MATCH "Mounted /dev/"
```
Comment (zyga) [inline tests/main/interfaces-udisks2/task.yaml]: This use of retry-tool is incorrect. The MATCH will be invoked on the output of retry-tool, not mount.  Please wrap it in sh -c: 
```suggestion
    retry-tool -n 3 --wait 1 sh -c "test-snapd-udisks2.udisksctl mount -b \"$device\" -t ext4 | MATCH 'Mounted /dev/'"
```

**Comment by sergiocazzolato** [comment]
@zyga thanks for the suggestion, the fix has been applied.

File: tests/main/interfaces-udisks2/task.yaml [hunk 3]
```diff
@@ -41,7 +41,9 @@ execute: |
 
     device="$(losetup -j "$FS_PATH" | cut -d: -f1)"
 
-    test-snapd-udisks2.udisksctl mount -b "$device" -t ext4 | MATCH "Mounted /dev/"
+    # We retry because there is a race with the device becoming
+    # registered by udisks2. The issue can be easy reproduced in ubuntu-20.04
```
Comment (stolowski) [inline tests/main/interfaces-udisks2/task.yaml]:
```suggestion
    # registered by udisks2. The issue can be easily reproduced on ubuntu-20.04
```

**Comment by anonymouse64** [comment]
Seems to have blocked on uc20 preparation taking too long :-/ I've restarted the job

**Pull request merged**